torchtune/utils/metric_logging.py

```diff
@@ -317,3 +317,127 @@ def close(self) -> None:
         if self._writer:
             self._writer.close()
             self._writer = None
+
+
+class CometLogger(MetricLoggerInterface):
+    """Logger for use w/ Comet (https://www.comet.com/site/).
+    For more information about arguments expected by Comet, see
+    https://www.comet.com/docs/v2/guides/experiment-management/configure-sdk/#for-the-experiment.
+
+    Args:
+        api_key (Optional[str]): Comet API key. It's recommended to configure the API Key from the environment.
+        workspace (Optional[str]): Comet workspace name. If not provided, uses the default workspace.
+        project (Optional[str]): Comet project name. Defaults to Uncategorized.
+        experiment_name (Optional[str]): The name for comet experiment to be used for logging.
+        experiment_key (Optional[str]): The key for comet experiment to be used for logging. Must be an alphanumeric
```

Comment:
what is this needed for?

Reply:
The `experiment_key` is used in two situations:

* When resuming a Comet Experiment, it is the ID of the Experiment to resume logging data to.
* When creating a new Comet Experiment, we generate a random ID by default. In various cases, it is useful to control the new ID (like having the same ID for the Comet Experiment and the Cloud provider training job).